modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilder.java

```diff
@@ -63,79 +67,148 @@ public ArtifactClassLoaderModelBuilder(File applicationFolder) {
+      pluginOptional = findMuleMavenPluginDeclaration();
+    }
     if (processSharedLibraries) {
-      exportSharedLibrariesResourcesAndPackages();
+      pluginOptional.ifPresent(plugin -> exportSharedLibrariesResourcesAndPackages(plugin));
+    }
+    if (processAdditionalPluginLibraries) {
+      pluginOptional.ifPresent(plugin -> processAdditionalPluginLibraries(plugin));
     }
     return super.build();
   }
 
-  private void exportSharedLibrariesResourcesAndPackages() {
-    doExportSharedLibrariesResourcesAndPackages();
-  }
-
-  /**
-   * Template method for exporting shared libraries and packages.
-   * By default, the pom needs to be parsed again to find which dependencies need to be shared.
-   */
-  protected void doExportSharedLibrariesResourcesAndPackages() {
-    Model model = getPomModelFolder(applicationFolder);
+  private Optional<Plugin> findMuleMavenPluginDeclaration() {
+    Model model = getPomModelFolder(artifactFolder);
     Build build = model.getBuild();
     if (build != null) {
       List<Plugin> plugins = build.getPlugins();
       if (plugins != null) {
-        Optional<Plugin> packagingPluginOptional =
-            plugins.stream().filter(plugin -> plugin.getArtifactId().equals(MULE_MAVEN_PLUGIN_ARTIFACT_ID)
-                && plugin.getGroupId().equals(MULE_MAVEN_PLUGIN_GROUP_ID)).findFirst();
-        packagingPluginOptional.ifPresent(packagingPlugin -> {
-          Object configuration = packagingPlugin.getConfiguration();
-          if (configuration != null) {
-            Xpp3Dom sharedLibrariesDom = ((Xpp3Dom) configuration).getChild("sharedLibraries");
-            if (sharedLibrariesDom != null) {
-              Xpp3Dom[] sharedLibraries = sharedLibrariesDom.getChildren("sharedLibrary");
-              if (sharedLibraries != null) {
-                for (Xpp3Dom sharedLibrary : sharedLibraries) {
-                  String groupId = getSharedLibraryAttribute(applicationFolder, sharedLibrary, "groupId");
-                  String artifactId = getSharedLibraryAttribute(applicationFolder, sharedLibrary, "artifactId");
-                  findAndExportSharedLibrary(groupId, artifactId);
+        return plugins.stream().filter(plugin -> plugin.getArtifactId().equals(MULE_MAVEN_PLUGIN_ARTIFACT_ID)
+            && plugin.getGroupId().equals(MULE_MAVEN_PLUGIN_GROUP_ID)).findFirst();
+
+      }
+    }
+    return empty();
+  }
+
+  private void exportSharedLibrariesResourcesAndPackages(Plugin packagingPlugin) {
+    doExportSharedLibrariesResourcesAndPackages(packagingPlugin);
+  }
+
+  private void processAdditionalPluginLibraries(Plugin packagingPlugin) {
+    doProcessAdditionalPluginLibraries(packagingPlugin);
+  }
+
+  protected void doProcessAdditionalPluginLibraries(Plugin packagingPlugin) {
+    Object configuration = packagingPlugin.getConfiguration();
+    if (configuration != null) {
+      Xpp3Dom additionalPluginDependenciesDom = ((Xpp3Dom) configuration).getChild("additionalPluginDependencies");
```

Comment:
We should move these strings to static variables. Maybe the best solution will be to add them in the mule-maven-plugin and import them from here.

Reply:
Sure, let's do that.